articles/devops-pipelines-automation-with-cloud-build-and-triggers/index.md

```diff
@@ -0,0 +1,414 @@
+```
+15. Return to Cloud Source Repositories and verify that your changes were added to source control.
+
+
+
+#### Automate builds with Triggers
+Here we will make some changes to the application and then automate the build using the `triggers`. This helps to update application versions.
+
+1. On the menu, navigate to `Container Registry`. At this point, you should have a folder named `devops-image` with at least one container in it.
+2. On the Navigation menu, click `Cloud Build`. The Build history page should open, and one or more builds should be in your history.
+3. Click the `Triggers` link on the left and then click on the `Create trigger` button.
+4. Name the trigger `devops-trigger`
+5. Select your `devops-repo` Git repository.
+6. Select `.*(any branch)` for the branch.
+7. Choose `Dockerfile` for Build Configuration and select the default image.
+
+Note:) The image below describes the way your build configurations will be.
+
+![Configuration](/devops-pipelines-automation-with-cloud-build-and-triggers/automate.png)
+
+8. Accept the rest of the defaults, and click `Create`.
+9. To test the trigger, click `Run` and then `Run trigger`.
+10. Click the `History` link and you should see a build running. Wait for the build to finish, and then click the link to it to see its details.
+11. Scroll down and look at the logs. The output of the build here is what you would have seen if you were running it on your machine.
+12. Return to the Container Registry service. You should see a new folder, `devops-repo`, with a new image in it.
+13. Return to the Cloud Shell Code Editor. Find the file main.py in the `learning-devops/devops-repo` folder.
+14. In the main() function, change the title property to `This is from build trigger.` as shown below:
+
+```Flask
+@app.route("/")
+def main():
+    model={"title": "This is from build trigger."}
+    return render_template('index.html', model=model)
+```
+16. Commit the change with the following command:
+```
+cd ~/learning-devops/devops-repo
+git commit -a -m "build trigger test
+```
+17. Enter the following to push your changes to Cloud Source Repositories:
+```
+git push origin master
+```
+18. Return to the Cloud Console and the Cloud Build service. You should see another build running.
+
+
+#### Testing Build Changes
+1. When the build completes, click on it to see its details. Under Execution Details, copy the image link, the format should be `gcr.io/<your-project-id>/devops-repoxx34345xx`.
+2. Go to the Compute Engine service. As you did earlier, create a new virtual machine to test this image. Click on the box to deploy a container image to the virtual machine, and paste the copied image.
+3. Select Allow HTTP traffic.
+4. When the new machine is created, you can test your changes by requesting the virtual machine's external IP address in your browser. Your new message would be displayed.
+
+Note:) You might have to wait a few minutes after the VM is created for the Docker container to start.
+
```

Comment:
Can we have all the codes in a GitHub repo?

It'll be easy for the readers to refer to.